File: EDAII x Java/Grafos/FloydWarshall.java
import java.util.List;
import java.util.LinkedList;

public class FloydWarshall{

	public static int INF = 9999;

	public static void main(String[] args){
		Grafo g = new Grafo(true,true);

		// Generando nodos
		Vertice va = new Vertice("a");
		Vertice vb = new Vertice("b");
		Vertice vc = new Vertice("c");
		Vertice vf = new Vertice("f");
		Vertice vi = new Vertice("i");
		Vertice vj = new Vertice("j");

		// Añadiendo nodos al grafo
		g.añadirVertice(va);
		g.añadirVertice(vb);
		g.añadirVertice(vc);
		g.añadirVertice(vf);
		g.añadirVertice(vi);
		g.añadirVertice(vj);

		// Generando aristas
		Arista ab = new Arista(va,vb);
		Arista ac = new Arista(va,vc);
		Arista bc = new Arista(vb,vc);
		Arista cb = new Arista(vc,vb);
		Arista bf = new Arista(vb,vf);
		Arista ci = new Arista(vc,vi);
		Arista ii = new Arista(vi,vf);
		Arista ij = new Arista(vi,vj);
		Arista fj = new Arista(vf,vj);

		// Conectando nodos
		g.añadirArista(ab,4);
		g.añadirArista(ac,2);
		g.añadirArista(bc,6);
		g.añadirArista(cb,6);
		g.añadirArista(bf,7);
		g.añadirArista(ci,3);
		g.añadirArista(ii,1);
		g.añadirArista(ij,5);
		g.añadirArista(fj,3);

		g.imprimirGrafo();

		floydWarshall(g);
	}

	public static void floydWarshall(Grafo g){
		int distancias[][] = new int[g.vertices.size()][g.vertices.size()];
		int i, j, k;

		// Inicializando matriz de adyacencia
		for (i=0; i<g.vertices.size(); i++){
			List<Arista> adyacentes = g.vertices.get(i).aristas;

			for (j=0; j<g.vertices.size(); j++){

				if (i == j){
					distancias[i][j] = 0;
				} else {
					Vertice uAct = g.vertices.get(j);

					Arista a = buscarAdyacencia(adyacentes, uAct);
					if (a != null){
						distancias[i][j] = a.costo;
					} else {
						distancias[i][j] = INF;
					}
				}
			}
		}

		// Obteniendo rutas mínimas para cada vertice
		for (k=0; k<g.vertices.size(); k++){
			for (i=0; i<g.vertices.size(); i++){
				for (j=0; j<g.vertices.size(); j++){
					if (distancias[i][k] + distancias[k][j] < distancias[i][j]){
						distancias[i][j] = distancias[i][k] + distancias[k][j];
					}
				}
			}
		}

		// Imprimiendo matriz de adyacencia resultante
		for (i=0; i<g.vertices.size(); i++){
			// Matriz de adyacencia
			System.out.println("Nodo: " + g.vertices.get(i).nombre + ", ");
			for (j=0; j<g.vertices.size(); j++){
				System.out.println("\tDistancia al nodo: " + g.vertices.get(j).nombre + " -> " + distancias[i][j]);
			}
			System.out.println("");
		}
	}

	public static Arista buscarAdyacencia(List<Arista> aristas, Vertice u){
		for (int i=0; i<aristas.size(); i++){
			if (aristas.get(i).u == u){
				return aristas.get(i);
			}
		}
		return null;
	}

}

File: EDAII x Python/Grafos/FloydWarshall.py
from Grafo import Grafo
from Arista import Arista
from Vertice import Vertice

class FloydWarshall:

	INF = 9999

	def __init__(self):
		'Constructor de la clase'
		pass

	def floydWarshall(self,g):
		'Determina la ruta más corta de un nodo origen hacia el resto, para todos los nodos'
		distancias = [[0 for j in range(len(g.vertices))] for i in range(len(g.vertices))]

		# Inicializando matriz de adyacencia
		for i in range(0,len(g.vertices)):
			adyacentes = g.vertices[i].aristas

			for j in range(0,len(g.vertices)):
				if i==j:
					distancias[i][j] = 0
				else:
					uAct = g.vertices[j]
					a = self.buscarAdyacencia(adyacentes, uAct)

					if a != None:
						distancias[i][j] = a.costo;
					else:
						distancias[i][j] = self.INF

		# Obteniendo rutas mínimas para cada vertice
		for k in range(0,len(g.vertices)):
			for i in range(0,len(g.vertices)):
				for j in range(0,len(g.vertices)):
					if distancias[i][k] + distancias[k][j] < distancias[i][j]:
						distancias[i][j] = distancias[i][k] + distancias[k][j]

		# Imprimiendo matriz de adyacencia resultante
		for i in range(0,len(g.vertices)):
			# Matriz de adyacencia
			print("Nodo: ", g.vertices[i].nombre)
			for j in range(0,len(g.vertices)):
				print("\tDistancia al nodo: ",g.vertices[j].nombre," -> ",distancias[i][j])
			print(" ")

	def buscarAdyacencia(self,aristas,u):
		'Verifica si un nodo es próximo a otro'
		for i in range(0,len(aristas)):
			if aristas[i].u == u:
				return aristas[i]
		return None

# Ejecutando los objetos
programa = FloydWarshall()
g = Grafo(True,True);

# Generando nodos
va = Vertice("a");
vb = Vertice("b");
vc = Vertice("c");
vf = Vertice("f");
vi = Vertice("i");
vj = Vertice("j");

# Añadiendo nodos al grafo
g.añadirVertice(va);
g.añadirVertice(vb);
g.añadirVertice(vc);
g.añadirVertice(vf);
g.añadirVertice(vi);
g.añadirVertice(vj);

# Generando aristas
ab = Arista(va,vb,0);
ac = Arista(va,vc,0);
bc = Arista(vb,vc,0);
cb = Arista(vc,vb,0);
bf = Arista(vb,vf,0);
ci = Arista(vc,vi,0);
ii = Arista(vi,vf,0);
ij = Arista(vi,vj,0);
fj = Arista(vf,vj,0);

# Conectando nodos
g.añadirArista(ab,4);
g.añadirArista(ac,2);
g.añadirArista(bc,6);
g.añadirArista(cb,6);
g.añadirArista(bf,7);
g.añadirArista(ci,3);
g.añadirArista(ii,1);
g.añadirArista(ij,5);
g.añadirArista(fj,3);

g.imprimirGrafo();

programa.floydWarshall(g)